blog_post.md

```diff
@@ -0,0 +1,144 @@
+
+![](http://i65.tinypic.com/xeliqs.png)
+
+```ruby
+# Tell Daru which variables are categorical
+shelter_data.to_category 'OutcomeType', 'AnimalType', 'SexuponOutcome', 'Breed', 'Color'
+
+# Or quantize a numerical variable to categorical
+shelter_data['AgeuponOutcome'] = shelter_data['AgeuponOutcome(Weeks)'].cut [0, 1, 4, 52, 260, 1500],
+    labels: [:less_than_week, :less_than_month, :less_than_year, :one_to_five_years, :more_than__five_years]
+
+# Do your operations on categorical data
+shelter_data['AgeuponOutcome'].frequencies.sort ascending: false
+```
+![](http://i67.tinypic.com/w1u3vs.png)
+
+```ruby
+small['Breed'].categories.size
+#=> 1380
+# Merge infrequent categories to make data analysis easy
+other_cats = shelter_data['Breed'].categories.select { |i| shelter_data['Breed'].count(i) < 10 }
+other_cats_hash = other_cats.zip(['other']*other_cats.size).to_h
+shelter_data['Breed'].rename_categories other_cats_hash
+shelter_data['Breed'].frequencies
+# View the data
+small['Breed'].frequencies.sort(ascending: false).head(10)
+```
+![](http://i64.tinypic.com/25rcu8m.png)
+
+Please refer to [this blog post](http://lokeshh.github.io/blog/2016/06/21/categorical-data/) to know more.
+
+
+#### Visualizing categorical data
+
+With the help of Nyaplot, Gnuplot and Gruff, Daru now provides ability to visualize categorical data as it does with orgdinary data.
+
+To plot vector with Nyaplot one needs to call the function `#plot`.
+
+```ruby
+# dv is a caetgorical vector
+dv = Daru::Vector.new ['III']*10 + ['II']*5 + ['I']*5, type: :category, categories: ['I', 'II', 'III']
+
+dv.plot(type: :bar, method: :fraction) do |p, d|
+  p.x_label 'Categories'
+  p.y_label 'Fraction'
+end
+```
+
+![](http://i64.tinypic.com/2s6onsw.png)
+
+Given a dataframe, one can plot the scatter plot such that the points color, shape and size can be varied acording to a categorical variable.
+
+```ruby
+# df is a dataframe with categorical variable :c
+df = Daru::DataFrame.new({
+  a: [1, 2, 4, -2, 5, 23, 0],
+  b: [3, 1, 3, -6, 2, 1, 0],
+  c: ['I', 'II', 'I', 'III', 'I', 'III', 'II']
+  })
+df.to_category :c
+
+df.plot(type: :scatter, x: :a, y: :b, categorized: {by: :c, method: :color}) do |p, d|
+  p.xrange [-10, 10]
+  p.yrange [-10, 10]
+end
+```
+
+![](http://i64.tinypic.com/2mcfx28.png)
+
+In a similar manner Gnuplot and Gruff also supports plotting of categorical variables.
+
+An additional work I did was to add Gruff with Daru. Now one do plotting of vectors and dataframes also using Gruff.
+
+See more notebooks on visualizing categorical data with Daru [here](http://nbviewer.jupyter.org/github/SciRuby/sciruby-notebooks/tree/master/Data%20Analysis/Plotting/).
+
+#### Regression support for categorical data
+
+Now categorical data is supported in the regression in Statsample and Statsample-GLM.
+
+Also there has been formual language introduced (like used in R and Patsy) to ease the task of regression.
```

Comment:
It would be better to reformulate this sentence as `A new formula language (like that used in R or Patsy) has been introduced to  ease the task of specifying regressions.`.